src/execution/execute.ts

```diff
@@ -357,31 +417,69 @@ function executeOperation(
     );
   }
 
-  const rootFields = collectFields(
+  const { fields: rootFields, patches } = collectFields(
     exeContext.schema,
     exeContext.fragments,
     exeContext.variableValues,
     rootType,
     operation.selectionSet,
   );
   const path = undefined;
+  let result;
 
   switch (operation.operation) {
     case OperationTypeNode.QUERY:
-      return executeFields(exeContext, rootType, rootValue, path, rootFields);
+      result = executeFields(
+        exeContext,
+        rootType,
+        rootValue,
+        path,
+        rootFields,
+        exeContext.errors,
+      );
+      break;
     case OperationTypeNode.MUTATION:
-      return executeFieldsSerially(
+      result = executeFieldsSerially(
         exeContext,
         rootType,
         rootValue,
         path,
         rootFields,
       );
+      break;
     case OperationTypeNode.SUBSCRIPTION:
       // TODO: deprecate `subscribe` and move all logic here
       // Temporary solution until we finish merging execute and subscribe together
-      return executeFields(exeContext, rootType, rootValue, path, rootFields);
+      result = executeFields(
+        exeContext,
+        rootType,
+        rootValue,
+        path,
+        rootFields,
+        exeContext.errors,
+      );
+  }
+
+  for (const patch of patches) {
+    const { label, fields: patchFields } = patch;
+    const errors: Array<GraphQLError> = [];
+
+    exeContext.dispatcher.addFields(
+      executeFields(
+        exeContext,
+        rootType,
+        rootValue,
+        path,
+        patchFields,
+        exeContext.errors,
```

Comment:
@robrichard  Why do patches on the root type reuse the original grouped field set's list of errors here?

This differs from the `addFields` call for subpatches further down.

Reply:
Changing this does not seem to trigger any failing tests.

Reply:
Fixed and added a test. thanks!